manic/repository_git.py

```diff
@@ -103,7 +103,25 @@ def _clone_repo(self, base_dir_path, repo_dir_name, verbosity):
         """
         cwd = os.getcwd()
         os.chdir(base_dir_path)
-        self._git_clone(self._url, repo_dir_name, verbosity)
+
+        # get url
+        url = self._url
+
+        # add port
```

Comment:
It would help if you added comments giving examples of what this is doing - both the addition of port and user name. Something along the lines of, "If self._url is XXX and self._port is YYY, then the following will result in url being ZZZ".